### PR TITLE
Set modsulator-app request timeouts, and prevent files from being del…

### PR DESCRIPTION
**Pull request by tingulfsen**
…eted upon the first modsulator job failure.

### CODE REVIEW

**Comment by tingulfsen** [comment]
Thanks to @jkeck for showing how to use the not exactly intuitive Faraday library to POST requests with timeouts.

This pull request supersedes #747, which will be deleted once this pull request is approved.

**Comment by tingulfsen** [comment]
Note that the changes in app/jobs/modsulator_job.rb tie in with the corresponding changes in sul-dlss/modsulator-app#5

**Comment by tingulfsen** [comment]
This pull request partially addresses issues #741 and #731 by extending the timeout on POST requests to the MODSulator server. It also switches over from using Faraday to Rest Client, because I was unable to figure out how to set timeouts with POST requests using Faraday (sad, I know...).

This is only a partial fix, because there are performance issues when large files are uploaded (particularly large XML files). The current workaround is to break the files up into smaller pieces (e.g. under 10M.

During a sprint later this year, we plan to change the processing of XML files such that you can upload a zip file of many small XML files (each corresponding to a single MODS record). This will enable a "round trip" with descriptive metadata download, which currently provides just such a zip file as its output.

**Comment by ndushay** [comment]
In your comment above, it says " It also switches over from using Faraday to Rest Client, because I was unable to figure out how to set timeouts with POST requests using Faraday (sad, I know...)."  - perhaps the cut and pasted text needs a small edit?

**Comment by tingulfsen** [comment]
Sorry I edited that comment.

**Comment by ndushay** [comment]
Not spending a ton of brainpower on this;  it looks okay to me in that "nothing-is-screaming-at-me" sort of way.  Shameless green!

**Comment by jmartin-sul** [comment]
👍 from me.  would be nice to have someone more familiar with faraday give it a look, but it looks fine to me.

the only thing i can think of is something to test that faraday respects timeout settings, but how to go about writing that test isn't immediately apparent to me.  wouldn't hold up merging over it.

File: app/jobs/modsulator_job.rb [hunk 1]
```diff
@@ -10,6 +10,9 @@ class ModsulatorJob < ActiveJob::Base
   # A somewhat easy to understand and informative time stamp format
   TIME_FORMAT = '%Y-%m-%d %H:%M%P'
 
+  # Wait 30 minutes for remote requests to complete
+  TIMEOUT = 1800
```
Comment (ndushay) [inline app/jobs/modsulator_job.rb]: in a perfect world, this would be in a Settings.yml file

**Pull request merged**